### PR TITLE
Merge Milestone-2 to Development

### PR DESCRIPTION
**Pull request by Blasco9**
In this branch, we created a mockup of the basic logic for the game, the board and the prompts for the users to play

### CODE REVIEW

**Comment by shshamim63** [comment]
## Status :space_invader: Changes Required :space_invader: 
___
Hi @Blasco9 and @AlfredoElizarraras :wave: ,
Great initiative. You have enclosed all of your messages inside a method. That is a great effort to keep you code DRY :clap: . But you have to add some additional messages to complete this milestone. So please follow the instructions below to complete this milestone.
___
### :spades: Mandatory Changes :spades: 
___
1.  game informs player if selected move is invalid. Suppose a player places a invalid position so We have to inform that user about the situation. Couldn't find it. So please add that message.
2. game informs player if selected move is a winning move. Which means a player must get notified if he is the winner. All I can see a message about game end. So have to add a proper message to describe the situation
3. game informs player if selected move is a draw move. Need to inform both user that it is a draw.
4. Instead of 
https://github.com/AlfredoElizarraras/tic-tac-toe/blob/e6d649ded5b6e4e2daa5bb817cab17ff511823e1/bin/main.rb#L33-L45
I would have made it more flexible rather than making it constant. Cause if you wish to increase the number of row and column in future(dynamic), then you will have to modify in here too. That is against the single responsibility principle. So I will suggest to do something like this.
```ruby
game_on = true

# initial instructions for player that will incur once.

while game_on
  # loop for each move
  
  if winner # and/or draw
    game_on = false
  end
end
```
___
#### Please fix the above mentioned issues and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)

File: bin/main.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      print 'Not a valid number, try again: '
+      p2_move = gets.chomp.to_i
+    end
+    update_board(p2_move, board, 'O')
+    print "\n"
+    if player_won?(2)
+      puts "Player 2 wins!\n\n"
+      break
+    end
+  end
```
Comment (stratospherique) [inline bin/main.rb]: * [x] Inside loop, all actions should be repeated for each player **separately**. It should make your code more DRY
- **[HINT]** To accomplish the above, you need to figure some sort of turn switching mechanism

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work implementing the suggested changes :clap: :clap: However, some requirements related to the game loop should be implemented before we approve the milestone. Check :arrow_down: 

- **[REQUIRED]** Check the code comments below, they should help you refactor your code
- **[OPTIONAL]** Ask players for their names, it makes your app more user-friendly 

### That's it :relieved: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Comment by shshamim63** [comment]
## Status :tada:  Approved :tada: 
___
Amazing job. You have all the messages to guide a user. :clap: . So nothing is blocking me to approve this milestone.
#### This :+1: is from me. You can go ahead.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by AlfredoElizarraras** [comment]
Thanks for your review and approved! we proceed to merge then with development.

**Pull request merged**